packages/utils/arrays.ts

```diff
@@ -3,7 +3,7 @@ export const unique = <T>(arr: T[]) => [...new Set(arr)]
 type Many<T> = T | ReadonlyArray<T>
 // TODO: rename to `ensureArray`
 /** like `_.castArray`, except falsy value returns empty array. */
-export const castArray = <T>(arr: Many<T>): T[] => {
+export const castArray = <T>(arr: Many<T>): (T & any[]) | Many<T>[] => {
```

Comment:
`castArray`  is desinged to return T[]. Refer to [this](https://github.com/element-plus/element-plus/pull/18331/commits/7922a03104cc2c77fb73dc7aaaf2429609c19eb4#diff-04ce761abc5d9cd0ec2b8237e72b4cfadc5a897cc18d50c2f0bed57d9b4b4fecR8)